code/datums/beam.dm

```diff
@@ -143,6 +143,44 @@
 	anchored = TRUE
 	var/datum/beam/owner
 
+/obj/effect/ebeam/laser
+	name = "laser beam"
+	desc = "A laser beam!"
+	alpha = 200
+	var/strength = EYE_PROTECTION_FLAVOR
+	var/probability = 20
+
+/obj/effect/ebeam/laser/Crossed(var/atom/movable/AM)
+	. = ..()
+	if(! (prob(probability) && ishuman(AM)) )
+		return
+	var/mob/living/carbon/human/moving_human = AM
+	var/laser_protection = moving_human.get_eye_protection()
+	var/rand_laser_power = rand(EYE_PROTECTION_FLAVOR, strength)
+	if(rand_laser_power > laser_protection)
+		//ouch!
+		INVOKE_ASYNC(moving_human, /mob/proc/emote, "pain")
+		visible_message(SPAN_DANGER("[moving_human] screams out in pain as \the [src] moves across their eyes!"), SPAN_NOTICE("Aurgh!!! \The [src] moves across your unprotected eyes for a split-second!"))
+	else
+		if(HAS_TRAIT(moving_human, TRAIT_BIMEX))
+			visible_message(SPAN_NOTICE("[moving_human]'s BiMex© personal shades shine for a moment as \the [src] passes over them."), SPAN_NOTICE("Your BiMex© personal shades shine for a moment as \the [src] passes over them."))
+			//drip = bonus balloonchat
+			moving_human.balloon_alert_to_viewers(SPAN_NOTICE("[moving_human]'s BiMex© personal shades shine for a moment as \the [src] passes over them."), SPAN_NOTICE("Your BiMex© personal shades reflect \the [src]."))
```

Comment:
this first message is way too long and span classes don't work in balloon chat iirc

Reply:
okie dokie

Reply:
fixed
![image](https://user-images.githubusercontent.com/53100513/210111592-46cdcadc-e021-41d7-9806-d3a508c4f823.png)